core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -699,12 +853,24 @@ private class FsApplicationAttemptInfo(
     endTime: Long,
     lastUpdated: Long,
     sparkUser: String,
-    completed: Boolean = true)
-  extends ApplicationAttemptInfo(
-      attemptId, startTime, endTime, lastUpdated, sparkUser, completed)
+    completed: Boolean = true,
+    val fileSize: Long = -1,
+    val instance: Long = -1)
```

Comment:
I don't see the point of `instance` and the `attemptCounter`.  It seems that it is eventually compared by a modified time (in `FshistoryProvider#getAppUI`), I'm not sure why you don't just use the modified time from the start.  At least, in the tests, I replaced it w/ the modified time and things still pass.